AppCenter/AppCenter/Internals/Storage/MSDBStorage.m

```diff
@@ -72,6 +54,45 @@ - (instancetype)initWithVersion:(NSUInteger)version filename:(NSString *)filenam
   return (self = [self initWithSchema:nil version:version filename:filename]);
 }
 
+- (int)initializeDatabaseWithSchema:(MSDBSchema *)schema version:(NSUInteger)version filename:(NSString *)filename {
+  BOOL newDatabase = ![MSUtility fileExistsForPathComponent:filename];
+  self.dbFileURL = [MSUtility createFileAtPathComponent:filename withData:nil atomically:NO forceOverwrite:NO];
+  self.maxSizeInBytes = kMSDefaultDatabaseSizeInBytes;
+
+  int result;
+  sqlite3 *db = [MSDBStorage openDatabaseAtFileURL:self.dbFileURL withResult:&result];
+  if (!db) {
+    return result;
+  }
+  self.pageSize = [MSDBStorage getPageSizeInOpenedDatabase:db];
+  NSUInteger databaseVersion = [MSDBStorage versionInOpenedDatabase:db result:&result];
+  if (result != SQLITE_OK) {
+    sqlite3_close(db);
+    return result;
+  }
+
+  // Create table.
+  if (schema) {
+    result = [MSDBStorage createTablesWithSchema:schema inOpenedDatabase:db];
+    if (result != SQLITE_OK) {
+      sqlite3_close(db);
```

Comment:
We need to log an error here now

Reply:
Already printed, no need to duplicate

Reply:
We probably need to move error log to the method that actually taking a request from a caller to get rid of any confusions. I fully understand the @jwallra 's comment, we can probably move the error message to this method unless it would be a duplication of error messages.

Reply:
Let's do not remove the log from the common function. The additional log has been added.